### PR TITLE
New AppBar component (v1) and default props changes

### PR DESCRIPTION
**Pull request by vmilan**
# Description

Shortcut: [#277277](https://app.shortcut.com/cartoteam/story/277277/components-appbar)
[sc-277277]

Ths PR includes:
- Common AppBar elements and styles 
- IconButton svg size fix
- Design request: icon change in ButtonGroup (split variation). Also exported the icon to use externally.
- Basic default styles for layer positioning in popper and popover.

### CODE REVIEW

File: packages/react-ui/src/theme/carto-theme.d.ts [hunk 1]
```diff
@@ -63,6 +66,7 @@ type CustomPalette = Modify<
     white: CustomShadesPaletteColor;
     black: CustomShadesPaletteColor;
     brand: CustomBrandPaletteColor;
+    custom: CustomPaletteColor;
```
Comment (VictorVelarde) [inline packages/react-ui/src/theme/carto-theme.d.ts:69]: I'm not sure if we need here 2 'custom palettes' here, Can't we just use the current brand color in the component @vmilan ?

File: packages/react-ui/src/theme/sections/components/surfaces.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+export const surfacesOverrides = {
+  // AppBar
+  MuiAppBar: {
+    styleOverrides: {
+      root: {
+        flexDirection: 'row',
+        alignItems: 'center',
+        justifyContent: 'space-between',
+        height: APPBAR_SIZE,
+        backgroundColor: commonPalette.custom.appBar,
```
Comment (VictorVelarde) [inline packages/react-ui/src/theme/sections/components/surfaces.js:15]: theme.palette.branding ?

File: packages/react-ui/src/theme/sections/palette.js [hunk 3]
```diff
@@ -191,5 +191,8 @@ export const commonPalette = {
   },
   qualitative: {
     ...baseColors.qualitative
+  },
+  custom: {
+    appBar: '#162945' // brand.navyBlue
```
Comment (VictorVelarde) [inline packages/react-ui/src/theme/sections/palette.js:196]: Then why can't we reuse it from brand object?

File: packages/react-ui/storybook/stories/organisms/AppBar.stories.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      </AppBar>
+    </>
+  );
+};
+
+const commonArgs = {
+  brandText: 'CARTO',
+  secondaryText: 'Some text',
+  showBurgerMenu: true,
+  brandLogo: <img src='/carto-logo.svg' alt='' />
```
Comment (VictorVelarde) [inline packages/react-ui/storybook/stories/organisms/AppBar.stories.js:96]: I think it's a nice idea to encapsulate this, 👍🏻

File: packages/react-ui/src/theme/carto-theme.d.ts (hunk 1 above)
Comment (vmilan) [reply packages/react-ui/src/theme/carto-theme.d.ts]: Done!

**Review by VictorVelarde — APPROVED** [review]
👍🏻

**Pull request merged**